services/web/server/src/simcore_service_webserver/exporter/formatters/formatter_v2.py

```diff
@@ -0,0 +1,224 @@
+        service_key = entry["key"]
+        service_version = entry["version"]
+        label = entry["label"]
+
+        service_data = await get_service(
+            app=app,
+            user_id=user_id,
+            service_key=service_key,
+            service_version=service_version,
+            product_name=product_name,
+        )
+
+        service_data_inputs = service_data["inputs"]
+        for service_input in service_data_inputs.values():
+            input_entry = InputsEntryModel(
+                service_alias=label,
+                service_name=service_data["name"],
+                service_version=service_data["version"],
+                input_name=service_input["label"],
+                # not present on the service
+                input_data_ontology_identifier="",
+                input_data_type=service_input["type"],
+                # this is optional
+                input_data_units=service_input.get("unit", ""),
+                # not always available
+                input_data_default_value=str(service_input.get("defaultValue", "")),
+            )
+            inputs.append(input_entry)
+
+        service_data_outputs = service_data["outputs"]
+        for service_output in service_data_outputs.values():
+            output_entry = OutputsEntryModel(
+                service_alias=label,
+                service_name=service_data["name"],
+                service_version=service_data["version"],
+                output_name=service_output["label"],
+                # not present on the service
+                output_data_ontology_identifier="",
+                output_data_type=service_output["type"],
+                # this is optional
+                output_data_units=service_output.get("unit", ""),
+            )
+            outputs.append(output_entry)
+
+    code_description = CodeDescriptionModel(**params_code_description)
+    code_description_params = CodeDescriptionParams(
+        code_description=code_description, inputs=list(inputs), outputs=list(outputs)
+    )
+
+    # writing SDS structure with process pool to avoid blocking
+    with ProcessPoolExecutor(max_workers=1) as pool:
+        return await asyncio.get_event_loop().run_in_executor(
+            pool,
+            write_sds_directory_content,
+            base_path,
+            submission_params,
+            dataset_description_params,
+            code_description_params,
+        )
+
+
+class FormatterV2(BaseFormatter):
+    """Formates into the SDS format"""
+
+    def __init__(self, root_folder: Path):
+        super().__init__(version="2", root_folder=root_folder)
+
+    @property
+    def code_folder(self) -> Path:
+        return self.root_folder / "code"
+
+    async def format_export_directory(self, **kwargs) -> None:
+        kwargs["manifest_root_folder"] = self.root_folder
+
+        self.code_folder.mkdir(parents=True, exist_ok=True)
+        formatter_v1 = FormatterV1(root_folder=self.code_folder, version=self.version)
```

Comment:
This was the initial design, I had no idea, what was needed (when this was started), and instances provided the more flexibility. Have not given it much thought since.